.github/workflows/publish_github_actions.yaml

```diff
@@ -0,0 +1,213 @@
+# Copyright (c) 2025 - 2025, Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
+
+name: Test Macaron Action (tutorials)
+
+on:
+  push:
+    paths:
+    - action.yaml
+  pull_request:
+    paths:
+    - action.yaml
+  workflow_dispatch:
+
+permissions:
+  id-token: write
+  attestations: write
+
+jobs:
+  # tutorial-commit-finder:
+  #   name: Analyzing and comparing different versions of an artifact
+  #   runs-on: ubuntu-latest
+  #   steps:
+  #     - uses: actions/checkout@v4
+
+  #     - name: Run Macaron (analyze arrow@1.3.0)
+  #       uses: ./
+  #       with:
+  #         package_url: 'pkg:pypi/arrow@1.3.0'
+  #         output_dir: 'macaron_output/commit_finder'
+
+  #     - name: Run Macaron (analyze arrow@0.15.0)
+  #       uses: ./
+  #       with:
+  #         package_url: 'pkg:pypi/arrow@0.15.0'
+  #         output_dir: 'macaron_output/commit_finder'
+
+  #     - name: Run Macaron (verify policy - has-hosted-build)
+  #       uses: ./
+  #       with:
+  #         policy_file: './tests/tutorials/commit_finder/has-hosted-build.dl'
+  #         output_dir: 'macaron_output/commit_finder'
+
+  tutorial-detect-malicious-package:
+    name: Detecting malicious packages
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
```

Comment:
Please pin all of the third-party Actions to the hash and add the version string as a comment. See how we do this in the rest of workflows: https://github.com/oracle/macaron/blob/8791027190d235c4acc4c6ae126327acd70fc6c8/.github/workflows/_build.yaml#L55